### PR TITLE
Support non-uniform dividers on ADC inputs

### PR DESCRIPTION
**Pull request by dron0gus**
S105 board has different dividers on CLT and IAT vs MAP and TPS channels.
Add option and board callback `getAnalogInputDividerCoefficient()` to support getting ADC input dividers from board helper instead of `engineConfiguration->analogInputDividerCoefficient`
Should not affect boards with no `EFI_NON_UNIFORM_ANALOG_DIVIDER` option

### CODE REVIEW

**Comment by rusefillc** [comment]
Another idea, no arguments for pros and cons:
__weak__ functions getCltDividerOverride getIatDividerOverride etc

File: firmware/hw_layer/sensors/hip9011_logic.h [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #endif
 
 #define HIP9011_BAND(bore)		(900 / (CONST_PI * (bore) / 2))
-#define HIP9011_DESIRED_OUTPUT_VALUE	5.0f
+#define HIP9011_ANALOG_OUTPUT_MAX		5.0f
```
Comment (rusefillc) [inline firmware/hw_layer/sensors/hip9011_logic.h:19]: huh how is this HIP change in this PR? should that be a separate PR?

**Comment by rusefillc** [comment]
can we/shall we split into three PRs with two preparations (hip and map)?

**Comment by dron0gus** [comment]
> Another idea, no arguments for pros and cons: **weak** functions getCltDividerOverride getIatDividerOverride etc

In this case ADC code needs to know what user (CLT, IAT) is trying to get value. This is not good in terms of modular programming.
Also (not in case of S105 with its fixed pin functions) ADC input channel divider is parameter of ADC input, not of a function that is connected to this input.

File: firmware/hw_layer/sensors/hip9011_logic.h (hunk 1 above)
Comment (dron0gus) [reply firmware/hw_layer/sensors/hip9011_logic.h:19]: https://github.com/rusefi/rusefi/pull/4939
https://github.com/rusefi/rusefi/pull/4940

File: firmware/hw_layer/adc/adc_inputs.h [hunk 2]
```diff
@@ -15,6 +15,13 @@
 #define SLOW_ADC_RATE 500
 #endif
 
+#if EFI_NON_UNIFORM_ANALOG_DIVIDER == FALSE
```
Comment (mck1117) [inline firmware/hw_layer/adc/adc_inputs.h:18]: I would prefer

```c++
__attribute__((weak)) float getAnalogInputDividerCoefficient(adc_channel_e) {
    return engineConfiguration->analogInputDividerCoefficient;
}
```

This will do the same thing, but requires no defines anywhere, just adding the function in the board's cpp file.

File: firmware/config/boards/s105/board.mk [hunk 3]
```diff
@@ -31,6 +31,9 @@ DDEFS += -DSTM32_USB_USE_OTG2=FALSE
 DDEFS += -DEFI_FILE_LOGGING=FALSE
 USE_FATFS = no
 
+# This board has different dividers on adc inputs
+DDEFS += -DEFI_NON_UNIFORM_ANALOG_DIVIDER=TRUE
```
Comment (mck1117) [inline firmware/config/boards/s105/board.mk:35]: If we use `__attribute__((weak))`, we don't need any defines to select whether to use it or not.

File: firmware/hw_layer/adc/adc_inputs.h (hunk 2 above)
Comment (dron0gus) [reply firmware/hw_layer/adc/adc_inputs.h:18]: Personally I don't like weak definition. As it sometimes cause strange issues that is hard to debug.
But, np, I'll rewrite it to use weak attribute.

Comment (mck1117) [reply firmware/hw_layer/adc/adc_inputs.h]: Hm, I've never had an issue with it like that. And in this case, LTO will inline it anyway when optimization is enabled.
